signal_scanner_bot/transport.py

```diff
@@ -4,74 +4,89 @@
 import subprocess
 import concurrent.futures
 
-import tweepy
+from peony import events
 import ujson
 
 from . import env
 from . import messages
-from . import twitter
 from . import signal
 
 log = logging.getLogger(__name__)
 
 
 ################################################################################
-# Stream listener
+# Twitter-to-Queue
 ################################################################################
-class Listener(tweepy.StreamListener):
-    """
-    Listener implementation for when a status matching the track is received.
-    """
+def _filter_hashtags(data, filter_hashtag_list):
+    for input_hashtag in data["entities"]["hashtags"]:
+        if input_hashtag["text"] in filter_hashtag_list:
+            return True
+    return False
 
-    def on_status(self, status: tweepy.Status):
-        if not env.STATE.LISTENING:
-            return
 
-        messages.process_twitter_message(status)
+async def twitter_to_queue():
+    log.info("Starting Twitter Event Stream")
+    stream_obj = env.CLIENT.stream.statuses.filter.post(
+        follow=",".join(env.TRUSTED_TWEETERS)
+    )
+    async with stream_obj as stream:
+        async for data in stream:
+            if events.on_connect(data):
+                log.info("Connected to the stream")
+            elif events.on_tweet(data):
+                if _filter_hashtags(data, env.RECEIVE_HASHTAGS):
+                    await messages.process_twitter_message(data)
 
 
 ################################################################################
-# Twitter-to-signal
+# Queue-to-Signal
 ################################################################################
-def _twitter_to_signal():
+def _queue_to_signal():
```

Comment:
I believe this function and the async one below it can be combined and simplified! Specifically that this one can become public and async (while popping items from the queue) and the public one below can pretty much be deleted. We may need to move the `env.STATE.STOP_REQUESTED` logic here too.